### PR TITLE
Fix flowChatUpdates torn Room snapshots

### LINKED ISSUE #398: [Bug] flowChatUpdates can terminate on a torn Room snapshot
## Bug description

`LocalChatDataSourceImpl.flowChatUpdates()` (`LocalChatDataSourceImpl.kt:150-194`) reads from two independently-invalidated Room flows without a consistent snapshot, exposing a torn-read race.

It `flatMapLatest`es over `chatDao.flowChatWithParticipants(chatId)` and, inside, `map`s over `messageDao.flowLastMessageInChat(chatId)`. The inner message flow re-emits independently of the captured outer `chatWithParticipants` snapshot. A sync `ChatUpdatedDelta` transaction (`LocalSyncDataSourceImpl.applyChatUpdatedDelta`) can atomically add a new participant *and* a message from that new participant. After commit, Room invalidates both table sets, but the inner message flow can re-emit a fresh last-message against the still-stale captured participant set.

When that fresh message's sender is the newly-added participant absent from the captured snapshot, `EntityMappers.toChat` → `toMessage` hits `error("Sender not found for message $id")` (`EntityMappers.kt:189`). The `.catch` at `LocalChatDataSourceImpl.kt:186-194` only handles `SQLiteException` and rethrows everything else, so the per-chat updates flow terminates permanently.

This race was noted during the #396 review (progress log) as a pre-existing pattern in `flowChatUpdates`, untouched by that work.

## Steps to reproduce

1. Open a chat and collect its updates (`flowChatUpdates`).
2. Trigger a sync delta that, in one transaction, adds a new participant to the chat and a message authored by that new participant.
3. Hit the window where the inner `flowLastMessageInChat` re-emits before the outer participant flow catches up.

## Expected behavior

The flow keeps emitting consistent snapshots; transient torn pairs (a message whose sender is not yet in the observed participant set) are skipped rather than crashing the flow.

## Actual behavior

`toMessage` throws `Sender not found`, the non-`SQLiteException` is rethrown by `.catch`, and the flow terminates permanently (chat screen stops receiving updates until re-subscription). Intermittent, write-timing-dependent.

## Suggested fix

The torn `(stale participants, fresh message)` pair is a concurrency artifact of combining two independently-invalidated flows — it never existed as a committed state in the database. So skip such an emission rather than letting it propagate or crash: when the last message's sender is not present in the current participant snapshot, treat the pair as a transient intermediate and don't emit (the next, consistent emission follows immediately once the participant flow catches up). The observer thus receives a consistent, uninterrupted stream.

This does **not** conflict with the persistence-boundary fail-fast contract (CLAUDE.md). That contract concerns the integrity of the source of truth — the database — and mandates failing fast when *persisted* data violates the participant invariant. Here the database is consistent; only the cross-flow `combine`/`flatMapLatest` join transiently observes a mismatched pair. Dropping that pair is recovering from a concurrency race, not hiding an inconsistency in the source of truth.

## Build flavor

All

## Affected layer

Data

## Suggested priority

medium

## Logs, screenshots, or stack traces

```
IllegalStateException: Sender not found for message <id>
  at …EntityMappers.toMessage(EntityMappers.kt:189)
  at …EntityMappers.toChat(...)
  at …LocalChatDataSourceImpl.flowChatUpdates$…(LocalChatDataSourceImpl.kt:180)
```

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #398

## Summary
Prevents `flowChatUpdates` from terminating when independently invalidated Room flows temporarily expose a fresh last message with a stale participant snapshot.

## Changes
- Detect apparent orphan last-message senders before mapping a chat update.
- Verify apparent orphans by re-reading participants and the last message in one Room transaction.
- Recover transient torn reads while surfacing persistent orphan data as handled `InvalidData` failures.
- Make last-message selection deterministic for equal timestamps with an `id DESC` tie-breaker.
- Add component tests for the normal path, persistent-orphan failure and recovery, and deterministic DAO ordering.

## Important Design Decision
The flow uses a fast path for already-consistent snapshots and only performs a transactional re-read when the captured participant set does not contain the last-message sender. This avoids an extra database read for normal emissions while distinguishing a transient cross-flow torn read from a real persisted inconsistency. Persistent orphans remain visible as handled errors instead of being silently skipped.

## Testing
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
Not applicable; no UI changes.

## Acceptance criteria
- [x] `flowChatUpdates` keeps emitting consistent snapshots instead of terminating on a transient torn participant/message pair.
- [x] A persistent last-message sender orphan is surfaced as a handled `LocalDataSourceError.InvalidData`.
- [x] The same flow subscription can recover and emit success after invalid data is repaired.
- [x] Normal chat updates continue to use the direct mapping path.

## Breaking Changes
None.

## Follow-ups
None.

### CODE REVIEW

File: app/src/main/java/timur/gilfanov/messenger/data/source/local/LocalChatDataSourceImpl.kt [hunk 1]
```diff
@@ -191,6 +194,32 @@ class LocalChatDataSourceImpl @Inject constructor(
...
+     *   participants and last message inside one `withTransaction`. If the orphan is gone it was a
+     *   transient torn read and emits [ResultWithError.Success]; if it persists it is a real
+     *   inconsistency and emits a handled [LocalDataSourceError.InvalidData] [ResultWithError.Failure],
+     *   keeping the flow alive and the error visible. Escalation (not a silent skip) is correct
+     *   because a real orphan would otherwise blank the chat forever.
+     *
+     * Note: with a single real Room DB the verify path's transient-recovery branch is not
+     * deterministically forceable through this public flow — a genuinely consistent DB always
+     * takes the fast path, and the only way to observe a persisting orphan is to seed a real one,
+     * which escalates. The recovery branch is therefore not unit-forceable here.
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/data/source/local/LocalChatDataSourceImpl.kt]: Do we need this comment?
```suggestion
```

File: app/src/main/java/timur/gilfanov/messenger/data/source/local/LocalChatDataSourceImpl.kt [hunk 2]
```diff
@@ -191,6 +194,32 @@ class LocalChatDataSourceImpl @Inject constructor(
...
+     *
+     * To discriminate without silencing real orphans:
+     * - **Fast path**: when the captured participant snapshot already contains the last message's
+     *   sender, map directly with no extra DB read.
+     * - **Verify path**: when an apparent orphan is seen, [readChatUpdateConsistently] re-reads
+     *   participants and last message inside one `withTransaction`. If the orphan is gone it was a
+     *   transient torn read and emits [ResultWithError.Success]; if it persists it is a real
+     *   inconsistency and emits a handled [LocalDataSourceError.InvalidData] [ResultWithError.Failure],
+     *   keeping the flow alive and the error visible. Escalation (not a silent skip) is correct
+     *   because a real orphan would otherwise blank the chat forever.
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/data/source/local/LocalChatDataSourceImpl.kt]: Do we need this much details here?
